modules/consensus/database.go

```diff
@@ -56,417 +86,254 @@ func openDB(filename string) (*setDB, error) {
-
-		// Initilize the consistency guards.
-		cg, err := tx.CreateBucketIfNotExists(ConsistencyGuard)
+	// Create the database buckets.
+	for _, bucket := range buckets {
+		_, err := tx.CreateBucketIfNotExists(bucket)
 		if err != nil {
 			return err
 		}
-		gs := cg.Get(GuardStart)
-		ge := cg.Get(GuardEnd)
-		// Database is consistent if both are nil, or if both are equal.
-		// Database is inconsistent otherwise.
-		if (gs != nil && ge != nil && bytes.Equal(gs, ge)) || gs == nil && ge == nil {
-			cg.Put(GuardStart, encoding.EncUint64(1))
-			cg.Put(GuardEnd, encoding.EncUint64(1))
-			return nil
-		}
-		return errDBInconsistent
-	})
-	return &setDB{db, true}, err
-}
-
-// startConsistencyGuard activates a consistency guard on the database. This is
-// necessary because the consensus set makes one atomic database change, but
-// does so using several boltdb transactions. The 'guard' is actually two
-// values, a 'GuardStart' and a 'GuardEnd'. 'GuardStart' is incremented when
-// consensus changes begin, and 'GuardEnd' is incremented when consensus
-// changes finish. If 'GuardStart' is not equal to 'GuardEnd' when
-// startConsistencyGuard is called, the database is likely corrupt.
-func (db *setDB) startConsistencyGuard() error {
-	return db.Update(func(tx *bolt.Tx) error {
-		cg := tx.Bucket(ConsistencyGuard)
-		gs := cg.Get(GuardStart)
-		if !bytes.Equal(gs, cg.Get(GuardEnd)) {
-			println("Database is inconsistent - please reset your database by redownloading it or loading a consistent backup. This can happen if you close Sia unexpectedly.")
-			return errDBInconsistent
-		}
-		i := encoding.DecUint64(gs)
-		return cg.Put(GuardStart, encoding.EncUint64(i+1))
-	})
-}
-
-// stopConsistencyGuard is the complement function to startConsistencyGuard.
-// startConsistencyGuard should be called any time that consensus changes are
-// starting, and stopConsistencyGuard should be called when the consensus
-// changes are finished. The guards are necessary because one set of changes
-// may occur over multiple boltdb transactions.
-func (db *setDB) stopConsistencyGuard() {
-	err := db.Update(func(tx *bolt.Tx) error {
-		cg := tx.Bucket(ConsistencyGuard)
-		i := encoding.DecUint64(cg.Get(GuardEnd))
-		return cg.Put(GuardEnd, encoding.EncUint64(i+1))
-	})
-	if err != nil && build.DEBUG {
-		panic(err)
 	}
-}
 
-// insertItem inserts an item to a bucket. In debug mode, a panic is thrown if
-// the bucket does not exist or if the item is already in the bucket.
-func insertItem(tx *bolt.Tx, bucket []byte, key, value interface{}) error {
-	b := tx.Bucket(bucket)
-	if build.DEBUG && b == nil {
-		panic(errNilBucket)
-	}
-	k := encoding.Marshal(key)
-	v := encoding.Marshal(value)
-	if build.DEBUG && b.Get(k) != nil {
-		panic(errRepeatInsert)
+	// Set the block height to -1, so the genesis block is at height 0.
+	blockHeight := tx.Bucket(BlockHeight)
+	underflow := types.BlockHeight(0)
+	err := blockHeight.Put(BlockHeight, encoding.Marshal(underflow-1))
```

Comment:
seems hackish. Perhaps better to just add the height + id manually, instead of using `pushPath`.